src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
```

Comment:
It's not so much the compiler, but the file system: what means include "once"? Is including the same file through different symbolic links prevented by this? What if the compiler can not detect that two files are the same in the file system?

See # 841.